### DISCUSSION TITLE
WARNING in error: Parsing error: Unexpected token < at

### DISCUSSION THREAD

**Discussion by code1x1** [webpack/webpack#14977] [General]
# Bug report

WARNING in error: Parsing error: Unexpected token < at  xxx.html:1:1:
> 1 | <script> </script>

**What is the current behavior?**

```
WARNING in error: Parsing error: Unexpected token < at app/atomic-components/molecules/forms/form-elements/product-price-domain/product-price-domain.html:1:1:
> 1 | <script> /** @type {MoleculeFormProductPriceDomainController} */ let $ctrl; </script>
    | ^
  2 | <div class="product-price-domain__wrapper">
  3 |     <div ng-if="!$ctrl.loadingPrice">
  4 |         <span ng-if="$ctrl.pricePrefixText" ng-class="{'space-left': !$ctrl.pricePrefixTextWithoutSpace}">{{ $ctrl.pricePrefixText }}</span>


error: Parsing error: Unexpected token < at app/atomic-components/molecules/lists/available-domains-list/available-domains-list.html:1:1:
> 1 | <script> /** @type {MoleculeListsAvailableDomainsListController} */ let $ctrl; </script>
    | ^
  2 | <section
  3 |     ng-if="$ctrl.loadingData"
  4 |     class="text-center space-top space-bottom"


2 errors found.

```

**If the current behavior is a bug, please provide the steps to reproduce.**

https://github.com/code1x1/webpack-unexpected-token

the config rule in question is located here:
https://github.com/code1x1/webpack-unexpected-token/blob/master/webpack/webpack.common.js#L76

this just worked long time now the config just triggers an error for every html file you save...

**What is the expected behavior?**

Error message should print which plugin/loader emited the error

**Other relevant information:**
webpack version:  5.65.0
Node.js version: v16.9.1
Operating System: Debian 10
Additional tools: npm v7.23.0

**Comment by alexander-akait** [comment]
There is not loader, it is not JS, so you don't see loaders/plugins

**Reply by alexander-akait** [reply]
No one handle it, so webpack try to parse it as JS

**Reply by code1x1** [reply]
it gets handled by html-loader https://github.com/code1x1/webpack-unexpected-token/blob/master/webpack/webpack.common.js#L79

**Reply by alexander-akait** [reply]
Do you enable `thread-loader`?

**Reply by code1x1** [reply]
doesnt matter you can uncomment this line and it still throws the error

**Reply by code1x1** [reply]
> Do you enable `thread-loader`?

what do you mean by that. we just used it like in webpack.js.org documentation described
https://webpack.js.org/loaders/thread-loader/

**Reply by alexander-akait** [reply]
`thread-loader` has some bugs and in most of case do not speed up build, I recommend do not use it

**Reply by code1x1** [reply]
Okay thought it would speed thinks up a little bit.
We have ~3000 component Templates which are bundled and since this loader has no cross dependency to any J's or typescript file we thought It would be good to use

**Reply by code1x1** [reply]
Anyway the second any of block throws the error above even though the html file gets bundled correct

**Reply by code1x1** [reply]
@alexander-akait the problem still exists if I remove thread-loader

**Reply by alexander-akait** [reply]
There is not loader/plugin error, webpack try to parse it as JS, but it is HTML, what do you expected, applied loaders before?

**Reply by code1x1** [reply]
but it should be handled by html-loader. then can you explain why the rule is not applied?
here is the `.html` rule => https://github.com/code1x1/webpack-unexpected-token/blob/master/webpack/webpack.common.js#L59
here is the `?external` oneOf block => https://github.com/code1x1/webpack-unexpected-token/blob/master/webpack/webpack.common.js#L68
and here is the fallback  block which should catch all other imports of html files => https://github.com/code1x1/webpack-unexpected-token/blob/master/webpack/webpack.common.js#L76

and this worked in wabpack 4

**Reply by alexander-akait** [reply]
Can you describe your problem? You need just do it works?

**Reply by code1x1** [reply]
it works but the console gets spammed with error messages and makes it hard to see real error messages

**Reply by code1x1** [reply]
> There is not loader/plugin error, webpack try to parse it as JS, but it is HTML, what do you expected, applied loaders before?

also if there is no loader the error message is something like that

`You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file.`

**Reply by alexander-akait** [reply]
I don't understand your problem, can you clarify? Webpack writes no loaders were applied, what you expected here?

```
ERROR in ./app/fact-sheet/fact-sheet.html 1:0
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```
